src/librustdoc/clean/inline.rs

```diff
@@ -395,7 +395,31 @@ crate fn build_impl(
             impl_
                 .items
                 .iter()
-                .map(|item| tcx.hir().impl_item(item.id).clean(cx))
+                .map(|item| tcx.hir().impl_item(item.id))
+                .filter(|item| {
+                    // Filter out impl items whose corresponding trait item has `doc(hidden)`.
```

Comment:
I think you need to add `if cx.document_hidden { return true; }`.

Reply:
Can you also add a test for // document-hidden-items at the same time?

Reply:
![2021-09-25-160757_7680x2160_scrot](https://user-images.githubusercontent.com/57856193/134762440-36c6aafc-e45d-4aae-934c-ea542925e799.png)
Checking if `--document-hidden-items` passed did the trick :100: